tools/sof-kernel-log-check.sh

```diff
@@ -246,32 +246,21 @@ case "$platform" in
         ;;
 esac
 
-[[ ! "$err_str" ]] && echo "Missing error keyword list" && exit 0
-# dmesg KB size buffer size
-#dmesg_config_define=$(awk -F '=' '/CONFIG_LOG_BUF_SHIFT/ {print $2;}' /boot/config-$(uname -r))
-#dmesg_buffer_size=$( echo $(( (1<<$dmesg_config_define) / 1024 )) )
-# kernel file log buffer size
-#kernel_buffer_size=$(du -k /var/log/kern.log |awk '{print $1;}')
-# now decide using which to catch the kernel log
-#[[ $kernel_buffer_size -lt $dmesg ]] && cmd="dmesg" || cmd="tail -n  +${begin_line}  /var/log/kern.log"
+# confirm begin_timestamp is in UNIX timestamp format, otherwise use dmesg
+journalctlflag="-k -q --no-pager --utc --output=short-iso --no-hostname"
+[[ $begin_timestamp =~ ^[0-9]{10} ]] && cmd="journalctl $journalctlflag --since=@$begin_timestamp" || cmd="dmesg"
```

Comment:
Why stick to `dmesg`? Its output looks very different. Why not just drop the `--since=` parameter?

Reply:
@marc-hb We need to keep dmesg as some user still use dmesg way. See https://github.com/thesofproject/sof-test/pull/468
commits like https://github.com/thesofproject/sof-test/pull/468/commits/8db6f8c18a81d79b082284bc452d18132e2552e2